### ISSUE TITLE
merge feat/fix-time-impurity

### ISSUE THREAD

**Issue by FlorianUekermann** [firezone/boringtun#41]
Is there a plan for maintaining a branch which contains all fixes/improvements? In particular I'm interested in the changes in `feat/fix-time-impurity`, on top of `master`.

**Comment by thomaseizinger** [comment]
Yes, the long-term plan is to definitely merge that branch back into `master`. The problem is that actually using that in our product (https://github.com/firezone/firezone) causes some issues with the test suite (which precisely controls time) that I need to fix first. What might help is doing https://github.com/firezone/boringtun/issues/19 first as well. I haven't had time to focus on that unfortunately.

In general, `master` of this fork is meant to contain all fixes.

If you are willing to, you can rebase that branch on top of `master` and we can maintain a 2nd long-lived branch next to master. Or you maintain your own fork of this fork? :)

**Comment by thomaseizinger** [comment]
I realised that we can just merge this PR on top of current `master`! Because it is backwards-compatible anyway, we can still update our codebose to the latest version and fix our test suite in parallel.
